src/Halcyon/HAL/HALResponse.cs

```diff
@@ -26,9 +25,9 @@ public class HALResponse {
         public HALResponse(object model, IHALModelConfig config = null)
             : this(config) {
             if (!(model is JObject) && (model is IEnumerable)) {
-                throw new ArgumentException("The HAL model should be Enumerable. You should use an embedded collection instead", "model");
+                throw new ArgumentException("The HAL model should be Enumerable. You should use an embedded collection instead", nameof(model));
             }
-
+            HALAttributeResolver.ResolveAttributes(this, model);
```

Comment:
In our case where we still need to modify the HALResponse (for pagination etc) the converter would not be able to find the underlying model, since the model would just by of type HALResponse. So therefore I need to be able to read the initial model where the attributes reside.